### ISSUE TITLE
.NET / Blazor support for Teams development

### ISSUE THREAD

**Issue by thomyg** [OfficeDev/TeamsFx#1215]
Congrats to the team for the new release! Really cool to see the effort you invest in Teams and how we, as developers, can extend and fine tune our customer's experience.

As I'm not a "React person" I'm keen to see your plans to bring the same type of tools to Visual Studio and/or the .NET world. As there are so many opportunities to use .NET for Teams development already, I'm sure you already think of ways to make things easier also for this group of developers. 

Happy to help in any way, shape, or form.

**Comment by adrianhall** [comment]
Hi @thomyg ,

Great news for you!  We have released a dotnet template for Blazor server, which includes new Fluent controls to ensure your app looks like it belongs in Teams, and we have a Visual Studio extension that implements the same.  Some links below for you:

Prerequisites: https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/prerequisites?tabs=vs
A tutorial: https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/first-app-blazor?tabs=vs
The dotnet new template NuGet: https://www.nuget.org/packages/Microsoft.TeamsFx.VisualStudio.ProjectTemplates/0.1.43-beta
The VSIX extension: https://marketplace.visualstudio.com/items?itemName=msft-vsteamstoolkit.vsteamstoolkit

I hope this helps!

**Comment by thomyg** [comment]
Hi @adrianhall

That's super cool. 

First thoughts after 5 minutes browsing through the code:

- The Teams SDK version is set to 1.6.0 with, current is 1.10.0. Can this be referenced to always get the current version? Thinking of https://unpkg.com/%40microsoft/teams-js/dist/MicrosoftTeams.min.js for example. 
- Any thought on using the Microsoft Graph Toolkit, and its auth library in the future? This could be an option to open auth up to fall back to the mgt MSAL provider. With that, you could create an app, that not only works in Teams but can be both, Teams and stand alone. That also works in WASM, as you can configure the mgt providers without a client secret. For example: 
![image](https://user-images.githubusercontent.com/10920072/119781360-37477300-becb-11eb-97e5-84070d7135f3.png)
- Any idea to get a class representation for the Teams context? Extending TeamsFx.cs with the most used initial things one could grab from the JavaScript objects. For example, like Mika does at Blazorade-Teams: https://github.com/Blazorade/Blazorade-Teams/blob/master/Blazorade.Teams/Interop/Context.cs

**Comment by adrianhall** [comment]
Thanks for the ideas! We're actively working on the Visual Studio experience.  I've filed these in our internal backlog, and I'll reach out separately so you can pass the ideas directly to us.

**Comment by guru98** [comment]
I agree with thomyg suggestion. Teams apps are, at the end, apps with a context in that context you can know that you are in a personal tab, channel tab, meeting tab without this info is almost un-useful. Authentication and easy access to a graph client is relevant for any app but the context is also really relevant.

**Comment by therealjohn** [comment]
Agreed! I've created a work item for us to track designing and adding a feature to provide context to the blazor app.

**Comment by StephenOTT** [comment]
Any movement on this?  The blazor examples seem to have been removed from the docs.

**Comment by eriolchan** [comment]
@therealjohn I also find the get started document of Blazor has been removed from docs. Can you provide more information for @StephenOTT ?

**Comment by therealjohn** [comment]
It seems the docs were hidden while they were updated to correctly match the changes made to the VS 2019 extension. I will investigate and get them back up. Sorry for the confusion!
